File: Reverse Rot/ReverseRot.java
/**
 * Problem: Reverse Rot
 * https://open.kattis.com/problems/reverserot
 *
 * @author   Martin Björling
 * @email    martinbjorling@gmail.com
 * @version  1.0
 * @since    2015-11-25
 */

import java.lang.Math;
//import java.lang.StringBuilder;
import java.util.Scanner;
import java.util.ArrayList;

public class ReverseRot {
	
	public static void main(String[] args) {

		Scanner sc = new Scanner(System.in);

		ArrayList<String> answers = new ArrayList<String>();

		String alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ_.";

		while (true) {
			String inputString = sc.nextLine();

			if (inputString.equals("0")) break;

			String[] inputArr = inputString.split(" ");

			String returnStr = "";

			int rotation = Integer.parseInt(inputArr[0]);

			for (int i = 0; i < inputArr[1].length(); i++) {
				char curr = inputArr[1].charAt(i);

				int currPos = alphabet.indexOf(curr);

				int newPos = -1;

				if (currPos > alphabet.length()-1-rotation) {
					newPos = currPos + rotation - alphabet.length();
					returnStr += Character.toString(alphabet.charAt(newPos));
				} else {
					newPos = currPos + rotation;
					returnStr += Character.toString(alphabet.charAt(newPos));
				}
			}

			answers.add(new StringBuilder(returnStr).reverse().toString());

		}

		for (int j = 0; j < answers.size(); j++) {
			System.out.println(answers.get(j));
		}	
		
	}
}

File: Reverse Rot/ReverseRot.php
<?php
/**
 * Problem: Reverse Rot
 * https://open.kattis.com/problems/reverserot
 *
 * @author   Martin Björling
 * @email    martinbjorling@gmail.com
 * @version  1.0
 * @since    2015-11-25
 */

$answers = [];

$alphabet = str_split("ABCDEFGHIJKLMNOPQRSTUVWXYZ_.");

while (true) {
	$inputString = stream_get_line(STDIN, 1024, PHP_EOL);

	if ($inputString == "0") break;

	$inputArr = explode(" ", $inputString);

	$stringArr = str_split($inputArr[1]);

	$returnStr = "";

	$rotation = $inputArr[0];

	for ($i = 0; $i < count($stringArr); $i++) {
		$curr = $inputArr[1][$i];

		$currPos = array_search($curr, $alphabet);

		$newPos = -1;

		if ($currPos > count($alphabet)-1-$rotation) {
			$newPos = $currPos + $rotation - count($alphabet);
			$returnStr .= $alphabet[$newPos];
		} else {
			$newPos = $currPos + $rotation;
			$returnStr .= $alphabet[$newPos];
		}
	}

	$answers[] = strrev($returnStr)."\n";
    
}

foreach ($answers as $value) {
	fwrite(STDOUT, $value);
}


?>